### ISSUE TITLE
Summing over wrong dimension

### ISSUE THREAD

**Issue by alecjacobson** [pressureless/linear_algebra#69]
From the gallery:

```
`I(X;Y)` = ∑_i ∑_j x_j p_i,j log₂(p_i,j/∑_k x_k p_i,k)

where

x ∈ ℝ^n
p ∈ ℝ^(m×n)
```
produces an incorrect loop range for `j`. For example, in python:

```python
    for i in range(1, len(p)+1):
        sum_1 = 0
        for j in range(1, len(p)+1):
```

This seems like a new issue. The old python code on the gallery is correct: https://i-heart-la.github.io/Supplementary_webpages/gallery/convex_optimization_208/index.html

**Comment by alecjacobson** [comment]
I think there's no reason to check for `sparse` here https://github.com/pressureless/linear_algebra/blob/master/iheartla/la_parser/codegen_numpy.py#L368
just if it's a matrix...

**Comment by alecjacobson** [comment]
Fixed in 597dd978ace8a6cad05f55ee8840f9809de82317 . Sorry I accidentally pushed to master. Let me know if this was mistaken.
